src/shared/bootspec.c

```diff
@@ -1392,13 +1392,141 @@ static void boot_entry_file_list(
                 *ret_status = status;
 }
 
+static int copy_cmdline(
+                char *cmdline,
+                char **ret) {
```

Comment:
per coding style only parameters that are output-only should be called `ret` or `ret_xyz`. This one is input+output, hence should not be named that way.